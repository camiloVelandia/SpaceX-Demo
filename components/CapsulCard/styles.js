import styled from "styled-components";

export const Container = styled.section`
  width: 100%;
  margin-bottom: 40px;
  border: 1px solid white;
  border-radius: 25px;
  padding: 30px 20px;
  @media (min-width: 992px) {
    width: 42%;
    padding: 40px 30px;
  }
`;

export const Info = styled.div`
  border: 1px solid white;
  padding: 40px 15px;
`;
export const Name = styled.div`
  text-transform:uppercase;
  font-weight:600;
  font-size:28px;
`;
export const Lists = styled.div`
  padding: 40px 15px;
  display:flex;
  flex-wrap:wrap;
  justify-content:space-around;
`;


export const Text = styled.p`
  font-size: 14px;
`;
export const Figure = styled.figure`
  font-size: 14px;
  max-width:224px;
  max-height:224px;
  height:100%;
  margin:0 auto;
  margin-top:60px;
  & img{
    height:100%;
    width:100%;
    object-fit:contain;
  }
`;

export const Ul = styled.ul`
  text-align: center;
  margin: 0;
  list-style-position: inside;
  padding: 0;
  margin-top: 30px;
  width: 100%;
  & h2 {
    text-transform: capitalize;
    font-size: 24px;
    margin: 0;
    padding-bottom: 10px;
  }
  & li {
    text-transform: capitalize;
    font-size: 14px;
  }
  @media (min-width: 400px) {
    width: 48%;
    text-align:left;
  }
`;

export const Status = styled.div`
  margin-top: 60px;
  & p {
    margin: 0 auto;
    display: flex;
    border: 2px solid #98e832;
    color: #98e832;
    text-transform: uppercase;
    font-size: 16px;
    padding: 10px 40px;
    width: fit-content;
  }
  @media (min-width: 768px) {
    & p{
    font-size: 24px;

    }
  }
`;
