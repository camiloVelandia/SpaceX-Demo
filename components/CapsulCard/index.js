import React from 'react';
import { Container, Status, Figure, Info,Name, Text, Ul, Lists } from "./styles";

const CapsulCard = () => {
  return (
    <Container>
      <Info>
      <Name>name capsule</Name>
        <Text>
          With the ability to lift into orbit over 54 metric tons (119,000
          lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew,
          luggage and fuel--Falcon Heavy can lift more than twice the payload of
          the next closest operational vehicle, the Delta IV Heavy, at one-third
          the cost.
        </Text>
        <Lists>
          <Ul>
            <h2>Specs</h2>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
          </Ul>
         
          <Ul>
            <h2>Specs</h2>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
          </Ul>
          <Ul>
            <h2>Specs</h2>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
          </Ul>
          <Ul>
            <h2>Specs</h2>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
            <li> height: 70m</li>
          </Ul>
        </Lists>
        <Figure>
          <img src={'./images/capsul.png'} /> 
        </Figure>
      </Info>

      <Status>
        <p>status: active</p>
      </Status>
    </Container>
  );
};

export default CapsulCard;