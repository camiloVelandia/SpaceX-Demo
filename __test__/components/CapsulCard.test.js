import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import CapsulCard from "../../components/CapsulCard";

describe("<CapsulCard/>", () => {
  const banner = mount(<CapsulCard />);
  test("render del componente banner", () => {
    expect(banner.length).toEqual(1);
  });
});

describe("banner snaaapshot", () => {
  test("Comprobar ui del banner", () => {
    const banner = create(<CapsulCard />);
    expect(banner.toJSON()).toMatchSnapshot();
  });
});
