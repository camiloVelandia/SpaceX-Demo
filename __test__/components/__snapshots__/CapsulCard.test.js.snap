// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner snaaapshot Comprobar ui del banner 1`] = `
<section
  className="styles__Container-pem3wt-0 hDuWTJ"
>
  <div
    className="styles__Info-pem3wt-1 iqHQBU"
  >
    <div
      className="styles__Name-pem3wt-2 dnRoKq"
    />
    <p
      className="styles__Text-pem3wt-4 hLWchL"
    />
    <div
      className="styles__Lists-pem3wt-3 hhNOA-d"
    >
      <ul
        className="styles__Ul-pem3wt-6 gSVifP"
      >
        <h2>
          Specs
        </h2>
        <li>
           capacity: 
        </li>
        <li>
           first flight: 
        </li>
        <li>
           type: 
        </li>
        <li>
           height: 
        </li>
        <li>
           feet: 
        </li>
      </ul>
      <ul
        className="styles__Ul-pem3wt-6 gSVifP"
      >
        <h2>
          thrusters
        </h2>
        <li>
           type: 
        </li>
        <li>
           amount: 
        </li>
        <li>
           fuel1: 
        </li>
        <li>
           fuel2: 
        </li>
        <li>
           isp: 
        </li>
        <li>
           pods: 
        </li>
      </ul>
    </div>
    <figure
      className="styles__Figure-pem3wt-5 igbLxp"
    >
      <img />
    </figure>
  </div>
  <div
    className="styles__Status-pem3wt-7 kqwPVp"
  >
    <p>
      status: 
    </p>
  </div>
</section>
`;
